lib/src/models/retry_event_model.dart

```diff
@@ -0,0 +1,26 @@
+import 'dart:async';
+
+/// Retries the `retryFunction` after a set `timeInterval` until `dispose` is called
+class RetryEventModel {
+  /// Time interval in milliseconds
+  Duration timeInterval;
+  void Function(Timer timer) retryFunction;
+
+  Timer? _timer;
+
+  RetryEventModel({
+    required this.timeInterval,
+    required this.retryFunction,
+  }) {
+    startTimer();
+  }
+
+  void startTimer() {
+    _timer = Timer.periodic(timeInterval, retryFunction);
+  }
```

Comment:
this probably shouldn't be public, considering that it doesn't cancel the currently running timer.